python/lsst/pipe/tasks/getRegionTimeFromVisit.py

```diff
@@ -0,0 +1,77 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+__all__ = ["GetRegionTimeFromVisitTask", "GetRegionTimeFromVisitConfig"]
+
+
+import lsst.pipe.base as pipeBase
+from lsst.pipe.base.utils import RegionTimeInfo
+from lsst.utils.timer import timeMethod
+
+
+class GetRegionTimeFromVisitConnections(pipeBase.PipelineTaskConnections,
+                                        dimensions={"instrument", "group", "detector"},
+                                        defaultTemplates={"coaddName": "goodSeeing", "fakesType": ""}):
+
+    dummy = pipeBase.connectionTypes.Input(
+        doc="Placeholder connection to ensure visit-group mapping is done, but "
+            "only for images we're processing. Use of a catalog dataset lets "
+            "us also control how late in the pipeline the task is run.",
+        name="{fakesType}{coaddName}Diff_diaSrcTable",
+        storageClass="DataFrame",
+        dimensions={"instrument", "visit", "detector"},
+    )
+    output = pipeBase.connectionTypes.Output(
+        doc="The region and time associated with this group's visit.",
+        name="regionTimeInfo",
+        storageClass="RegionTimeInfo",
+        dimensions={"instrument", "group", "detector"},
+    )
```

Comment:
Won't that break things, at least once we have snaps? Every single catalog dataset in the pipeline is visit-level, not visit+exposure level.

Reply:
Right, adding `exposure` to some existing catalog dataset like the one you've defaulted is definitely not good, as it will break when we have snaps.  But a _second_ dummy dataset with `exposure` dimensions would be fine; this could be:

- an `Input` that defaults to `raw` (or `postISRCCD`, but `raw` seems safer);
- another `Output` that never gets written.

in either case, you'd set `multiple=True` to make it keep working when we have snaps.  The QG generation queries will end up joining `visit` to `group` via two different `exposures`, but that should all be transparent.

I lean towards the input to avoid accidentally registering the dummy dataset type.

Reply:
Fictitious links to the main pipeline make me a bit nervous over potential side effects...

> I lean towards the input to avoid accidentally registering the dummy dataset type.

What do you mean by this? I assumed that in the output case you would have to (deliberately) register the dummy dataset type for graph generation to work.

Reply:
Output dataset types do not have to be registered before QG generation.  The accident is just users being too liberal with the `--register-dataset-types` argument to `pipetask`, so if you're more comfortable with a dummy output, you're welcome to go with that.